bin/varnishd/cache/cache_obj.c

```diff
@@ -265,8 +265,10 @@ ObjExtend(struct worker *wrk, struct objcore *oc, ssize_t l, int final)
  */
 
 uint64_t
-ObjWaitExtend(const struct worker *wrk, const struct objcore *oc, uint64_t l)
+ObjWaitExtend2(const struct worker *wrk, const struct objcore *oc, uint64_t l,
+    enum boc_state_e *statep)
```

Comment:
I think it would be much simpler to change `ObjWaitExtend()` that has a single call site in tree.

Reply:
Last time I suggested a breaking changes in #4035 I received concerns, so this time I suggested the other way. But actually I agree that changing `ObjWaitExtend()` is the better option.

Reply:
I have changed the PR accordingly, squashed commits, improved commit messages, updated changes.rst and vrt.h and updated the PR description.